core/rawdb/accessors_chain.go

```diff
@@ -373,3 +373,28 @@ func FindCommonAncestor(db DatabaseReader, a, b *types.Header) *types.Header {
 	}
 	return a
 }
+
+// ReadShardState retrieves sharding state
+func ReadShardState(db DatabaseReader, hash common.Hash, number uint64) types.ShardState {
+	data, _ := db.Get(shardStateKey(number, hash))
+	if len(data) == 0 {
+		return nil
+	}
+	shardState := types.ShardState{}
+	if err := rlp.DecodeBytes(data, &shardState); err != nil {
+		log.Error("Fail to decode sharding state", "hash", hash, "number", number, "err", err)
+		return nil
+	}
+	return shardState
+}
+
+// WriteShardState stores sharding state into database
+func WriteShardState(db DatabaseWriter, hash common.Hash, number uint64, shardState types.ShardState) {
```

Comment:
why not return error?

Reply:
I follow the patterns for other Write function, all of the Write functions here not return error.  In my opinion, return an error might be better.